### PR TITLE
review: delay buffer

### PR DESCRIPTION
**Pull request by gzeoneth**
wip review, please see inline comments

### CODE REVIEW

File: src/bridge/DelayBuffer.sol [hunk 1]
```diff
@@ -192,7 +200,9 @@ library DelayBuffer {
         uint64 maxBuffer,
         uint64 amountPerPeriod,
         uint64 period,
-        uint64 roundOff
+        uint64 roundOff 
+        // @review would rather blend (amountPerPeriod, period) together to a fixed unit
+        // instead of needing to keep track of 3 terms (amountPerPeriod, period, roundOff)
```
Comment (gzeoneth) [inline src/bridge/DelayBuffer.sol:205]: there would still be rounding error but should be small enough to be negligible
we can always round down if we care about sequencer abusing the rounding

File: src/bridge/DelayBuffer.sol [hunk 2]
```diff
@@ -130,6 +130,7 @@ library DelayBuffer {
         uint64 blockNumber,
         uint64 timestamp
     ) internal {
+        // @review we can also prevent this by limiting the value you can set as threshold
```
Comment (shotaronowhere) [inline src/bridge/DelayBuffer.sol:133]: the thought is thresholdBlocks = type(uint64).max and thresholdSeconds = type(uint64).max effectively disables the buffer, so governance could turn off the buffer feature without upgrading the contract.

_setBufferConfig calls updateSyncValidity so saturation at type(unit64).max is handled

File: src/bridge/DelayBuffer.sol (hunk 1 above)
Comment (shotaronowhere) [reply src/bridge/DelayBuffer.sol:205]: you mean something like setting the replenish rate in the form of 1/period ? Or even a simplification which works for both blocks and seconds like if period = 12, then there is no round off.

Comment (gzeoneth) [reply src/bridge/DelayBuffer.sol:205]: yes, to have sth like
```
BASIS = 10000
amountPerSecondInBasis = 1000 # equals to 0.1s per 1 second
```
then when we calculate how much to replenish, we always around off. The rounding should be small enough that is negligible, and always work against the sequencer (i.e. sequencer cannot gain more delay buffer by abusing the rounding)

File: src/bridge/AbsBridge.sol [hunk 3]
```diff
@@ -127,6 +127,7 @@ abstract contract AbsBridge is Initializable, DelegateCallAware, IBridge {
         if (afterDelayedMessagesRead > 0) {
             delayedAcc = delayedInboxAccs[afterDelayedMessagesRead - 1];
         }
+        // @review should use accumulateSequencerInbox here
```
Comment (gzeoneth) [inline src/bridge/AbsBridge.sol:130]: this was not fixed in https://github.com/OffchainLabs/nitro-contracts/pull/160 yet

Comment (shotaronowhere) [reply src/bridge/AbsBridge.sol:130]: I implemented your suggestion when both sync and delay proofs were implemented.

when removing the sync proof, the auxiliary helper functions in the Messages.sol library were no longer used and removed. Therefore there is no longer an accumulateSequencerInbox function and hence the bridge was left unchanged.

**Pull request closed without merging**